src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -83,7 +83,9 @@ public Participation init(Exercise exercise, String username) {
         // common for all exercises
         // Check if participation already exists
         Participation participation = participationRepository.findOneByExerciseIdAndStudentLogin(exercise.getId(), username);
-        if (!Optional.ofNullable(participation).isPresent() || (!(exercise instanceof QuizExercise) && participation.getInitializationState() == ParticipationState.FINISHED)) { //create a new participation only if it was finished before (not for quiz exercise)
+        if (!Optional.ofNullable(participation).isPresent() ||
+            (!(exercise instanceof QuizExercise || exercise instanceof ModelingExercise) && participation.getInitializationState() == ParticipationState.FINISHED)) {
+            // create a new participation only if it was finished before (not for quiz exercise)
```

Comment:
this code gets a bit complex now, also when we want to handle new exercise types. Can we not say '

`if(... || (exercise instanceof ProgrammingExercise && participation.getInitializationState() == ParticipationState.FINISHED))`